packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -233,9 +246,78 @@ export class CXLMarketingNavElement extends LitElement {
       }, 1000)
     );
 
+    document.body.addEventListener('vaadin-overlay-open', this._onOverlayOpen.bind(this));
+    document.body.addEventListener('click', this._onBackBtnClick.bind(this));
+
     super.firstUpdated(changedProperties);
   }
 
+  /**
+   * Listen to overlay open event,
+   * store the overlay element if it has `.backbutton`.
+   *
+   * @since 2022.11.14
+   * @see https://app.clickup.com/t/3rgnekt
+   * @private
+   */
+  _onOverlayOpen(oe) {
+    const overlay = oe.target;
+
+    if (!overlay) {
+      return;
+    }
+
+    const backBtn = overlay.querySelector('.back-button');
+
+    if (!backBtn) {
+      return;
+    }
+
+    this.submenuOverlay = overlay;
+  }
+
+  /**
+   * Listen to back button click event,
+   * close the relevant overlay.
+   *
+   * @since 2022.11.14
+   * @see https://app.clickup.com/t/3rgnekt
+   * @private
+   */
+  _onBackBtnClick(e) {
+    if (!this.submenuOverlay || !this._eventBelongsToElement(e, 'vaadin-button', 'back-button')) {
+      return;
+    }
+
+    this.submenuOverlay.dispatchEvent(new CustomEvent('backBtnClose', { bubbles: true }));
+  }
+
+  /**
+   * Check if event is initiated by relevant element.
+   *
+   * @since 2022.11.14
+   * @param e
+   * @private
+   */
+  // eslint-disable-next-line class-methods-use-this
+  _eventBelongsToElement(e, name = '', class_name = '') {
+    const path = e.composedPath ? e.composedPath() : e.path || [];
+
+    const element = Array.from(path).filter((node) => {
+      if (!node || !node.nodeName || !node.classList) {
+        return false;
+      }
+
+      return node.nodeName.toLowerCase() === name && node.classList.contains(class_name);
+    });
+
+    if (!element || !element.length) {
+      return false;
+    }
+
+    return element;
+  }
+
```

Comment:
@saas786 Not that I know of. I found this, though it doesn't seem to be maintained (https://www.npmjs.com/package/event-within-element/v/0.0.2).

Reply:
For historical purposes:

https://github.com/JJWesterkamp/event-delegation
https://github.com/thednp/event-listener

---

> @saas786 Not that I know of. I found this, though it doesn't seem to be maintained (https://www.npmjs.com/package/event-within-element/v/0.0.2).

Can't seem to find the GH repo for it, and though name suggests it might the one, but its not maintanined and too old :)

---

> This may mean it's the wrong architecture to build in the first place. Has this strategy been discussed on Vaadin Discord?

If we can't find an alternative, this is solid approach, I have been using it specifically for even bubling purposes, if can't attach event listener to some element, can listen to parent and see if it belongs to certain element in path, 
but otherwise yeah, its not ideal architecture, so ultimately refactored it out.

I forgot the Lit & Vaadin moved to discord, so did some research on slack but found no relevant thread.
Andrew let me know they moved to discord, but by the time we implemented a sane solution.